Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -233,31 +233,70 @@ extension _ChatChannelListVC: _ChatChannelListControllerDelegate {
         _ controller: _ChatChannelListController<ExtraData>,
         didChangeChannels changes: [ListChange<_ChatChannel<ExtraData>>]
     ) {
-        let newChannelsCount = controller.channels.count
-        var movedItems: [IndexPath] = []
-        
+        var allIndexes = Set<IndexPath>()
+        var moveIndexes = Set<IndexPathMove>()
+        var insertIndexes = Set<IndexPath>()
+        var removeIndexes = Set<IndexPath>()
+        var updateIndexes = Set<IndexPath>()
+
+        // The verification of conflicts is just a temporary solution
+        // until the root cause of the conflicts has been solved.
+        var hasConflicts = false
+        let verifyConflict = { (indexPath: IndexPath) in
+            let (inserted, _) = allIndexes.insert(indexPath)
+            hasConflicts = !inserted || hasConflicts
+        }
+
+        for change in changes {
+            if hasConflicts {
+                break
+            }
+
+            switch change {
+            case let .insert(_, index):
+                verifyConflict(index)
+                insertIndexes.insert(index)
+            case let .move(_, fromIndex, toIndex):
+                verifyConflict(fromIndex)
+                verifyConflict(toIndex)
+                moveIndexes.insert(IndexPathMove(fromIndex, toIndex))
+            case let .remove(_, index):
+                verifyConflict(index)
+                removeIndexes.insert(index)
+            case let .update(_, index):
+                verifyConflict(index)
+                updateIndexes.insert(index)
+            }
+        }
+
+        if hasConflicts {
+            collectionView.reloadData()
+            channelsCount = controller.channels.count
```

Comment:
this has to be the other way around - first update the count, then reload the view